src/include/miopen/tensor.hpp

```diff
@@ -173,6 +175,30 @@ struct TensorDescriptor : miopenTensorDescriptor
 
     std::string ToString() const;
 
+    template <class Vector, class Op>
+    inline std::vector<int64_t> sort_permutation(const Vector& data, Op op) const
+    {
+        std::vector<std::int64_t> result(data.size());
+        std::iota(result.begin(), result.end(), 0);
+        std::sort(
+            result.begin(), result.end(), [&](auto x, auto y) { return op(data[x], data[y]); });
+        return result;
+    }
+
+    std::string GetLayout(std::string labels) const
+    {
+        if(labels.size() != strides.size())
+        {
+            MIOPEN_THROW(
+                "Invalid labels size. Layout labels size must be equavalent to stride size");
+        }
+
+        auto result = labels;
```

Comment:
It seems like assignment is useless.

Reply:
This is the solution I copy-pasted from Paul's implementation. It copy constructed the `result` from `lables` (instead the alternative is to `push_back()` in the transform stage), I assume it is slightly faster than the alternative.

Reply:
Yes, this is to avoid calling `push_back` as incrementing a pointer should be faster.

Reply:
Please add a comment that explains this. Otherwise the code looks like an error.

Reply:
Done in f58abdaa7aa8b69d840066b9b20cae8ef58dc75c